data/transactions/logic/simulate_test.go

```diff
@@ -487,6 +499,62 @@ func TestSimpleAppCall(t *testing.T) {
 	require.Empty(t, result.FailureMessage)
 }
 
+func TestRejectAppCall(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	l := makeSimulationTestLedger()
+	s := v2.MakeSimulator(l)
+
+	accounts := makeTestAccounts()
+	sender := accounts[0].Address
+
+	// Compile AVM program
+	ops, err := AssembleString(rejectAVMProgram)
+	require.NoError(t, err, ops.Errors)
+	prog := ops.Program
+
+	// Create program and call it
+	futureAppID := 1
+	txgroup := []transactions.SignedTxn{
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.ApplicationCallTx,
+				Header: makeBasicTxnHeader(sender),
+				ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+					ApplicationID:     0,
+					ApprovalProgram:   prog,
+					ClearStateProgram: prog,
+					LocalStateSchema: basics.StateSchema{
+						NumUint:      0,
+						NumByteSlice: 0,
+					},
+					GlobalStateSchema: basics.StateSchema{
+						NumUint:      0,
+						NumByteSlice: 0,
+					},
+				},
+			},
+		},
+		{
+			Txn: transactions.Transaction{
```

Comment:
nit: it appears sufficient to use just a single txn in this group, since they both will fail